docker/docker-compose.yml

```
@@ -0,0 +1,13 @@
+version: "3.9"
+services:
+  ipfs-host:
+    image: "ipfs/go-ipfs:latest"
+    environment:
+      IPFS_PROFILE: server
+    ports:
+      - "127.0.0.1:4001:4001"
+      - "127.0.0.1:5001:5001"
```

Is it necessary to provide `127.0.0.1`? Could it work with just port mapping?
```suggestion
      - 4001:4001
      - 5001:5001
```

---

Good question. Yesterday I was trying to host a file via IPFS Docker container and it was not available via IPFS gateway immediately. So I tried putting `127.0.0.1` as is shown in [IPFS Docker documentation](https://hub.docker.com/r/ipfs/go-ipfs) - `docker run -d --name ipfs_host -v $ipfs_staging:/export -v $ipfs_data:/data/ipfs -p 4001:4001 -p 127.0.0.1:8080:8080 -p 127.0.0.1:5001:5001 ipfs/go-ipfs:latest`. Before that I was using only ports everywhere, and when I could not get the file I started using it with the IP address. I will do more tests without having IP and see if that works. Now I know that IPFS can be kind of slow when you are only hosting a file from one location (my laptop) and it may take some time for gateways to pick it up - the "solution" I found for this is to try several different public gateways and hope one of them will get to my node and expose the file.

---

Did another test, and `127.0.0.1` is indeed not needed - removing it.